### PR TITLE
Fix Helm chart repository name

### PR DESCRIPTION
**Pull request by IrvingMg**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
Since the new staging registry provides a repo for charts, we don't need to store the chart in the same repo as the Kueue image anymore.

#### Which issue(s) this PR fixes:

Relates to #2377, #2476

#### Does this PR introduce a user-facing change?

```release-note
Support for helm charts in the us-central1-docker.pkg.dev/k8s-staging-images/charts repository
```

### CODE REVIEW

**Comment by mimowo** [comment]
@IrvingMg could you clarify what is the status of the PR? I'm a bit lost. The annotation is "cleanup", the title "Fix Helm chart repository name" suggests it is a bugfix, and the release-note "Support for helm charts in the us-central1-docker.pkg.dev/k8s-staging-images/charts repository" suggests this is a new feature.

File: Makefile [hunk 1]
```diff
@@ -32,12 +32,12 @@ DOCKER_BUILDX_CMD ?= docker buildx
 IMAGE_BUILD_CMD ?= $(DOCKER_BUILDX_CMD) build
 IMAGE_BUILD_EXTRA_OPTS ?=
 # TODO(#52): Add kueue to k8s gcr registry
-STAGING_IMAGE_REGISTRY := us-central1-docker.pkg.dev/k8s-staging-images/kueue
-IMAGE_REGISTRY ?= $(STAGING_IMAGE_REGISTRY)
+STAGING_IMAGE_REGISTRY := us-central1-docker.pkg.dev/k8s-staging-images
```
Comment (mimowo) [inline Makefile:35]: I see the analogous line``STAGING_IMAGE_REGISTRY := us-central1-docker.pkg.dev/k8s-staging-images/kueue` in Makefile-test.mk. Shouldn't we align it?

**Comment by mimowo** [comment]
> I've updated to bugfix. This PR is related to #2377, #2476. It is not causing an error, but it's pushing the Helm chart to the wrong repo.

I see, thanks for explaining!.

File: Makefile (hunk 1 above)
Comment (IrvingMg) [reply Makefile:34]: Done.

**Comment by mimowo** [comment]
/lgtm
/approve 
Thanks

**Pull request merged**